### ISSUE TITLE
Training IOU, Cass, OBj, etc. = nan ? 

### ISSUE THREAD

**Issue by kdupreez** [AlexeyAB/darknet#600]
during training on Yolo V3, I am getting "not a number" (nan) errors from console output??

Region 82 Avg IOU: -**nan**, Class: -**nan**, Obj: -**nan**, No Obj: 0.504497, .5R: -**nan**, .75R: -**nan**,  count: 0
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.477067, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: 0.329275, Class: 0.472234, Obj: 0.503358, No Obj: 0.493445, .5R: 0.347826, .75R: 0.000000,  count: 2

**Comment by AlexeyAB** [comment]
https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

> Note: If during training you see `nan` values in some lines then training goes well, but if `nan` are in all lines then training goes wrong.

**Comment by kdupreez** [comment]
Yes, something did go wrong in training.. but no idea what??  on YoloV2 VOC based training works 100% with same data set, only difference now is I am using Yolo V3..

Its 1 class.. i changed the classe=1 and filters=18..  But doesnt matter what I use for  batch and subdiv.. I always get these NaN and Out Of Memory exceptions..

**Comment by AlexeyAB** [comment]
> Region 82 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.504497, .5R: -nan, .75R: -nan, count: 0
> Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.477067, .5R: -nan, .75R: -nan, count: 0
> Region 106 Avg IOU: 0.329275, Class: 0.472234, Obj: 0.503358, No Obj: 0.493445, .5R: 0.347826, .75R: 0.000000, count: 2

There is nothing wrong. Training is going well.

----

https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

> change line batch to batch=64
> change line subdivisions to subdivisions=8

https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> if error Out of memory occurs then in .cfg-file you should increase subdivisions=16, 32 or 64: link

**Comment by kdupreez** [comment]
Thanks very much - I'll give that a shot.

**Comment by kdupreez** [comment]
OK, used Batch=32 and Subdisions=16 and that seems to work well with low memory usage

Thank you - feel free to close this issue.

**Comment by ZJJTSL** [comment]
I have got the same problem . The value of IOU class Obj is always nan ,but the noObj is not nan ,the same as loss,loss is not nan . I've tried Batch=32 Subdisions=16 it shows cuda run out. so what should i do now

**Comment by AlexeyAB** [comment]
@ZJJTSL 

> but the noObj is not nan ,the same as loss**,loss is not nan**

This is normal: https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

> **Note:** If during training you see `nan` values for avg (loss) field - then training goes wrong, but if `nan` is in some other lines - **then training goes well.**
